TAO/TAO_IDL/include/ast_interface_fwd.h

```diff
@@ -80,10 +80,23 @@ class TAO_IDL_FE_Export AST_InterfaceFwd : public virtual AST_Type
   virtual ~AST_InterfaceFwd (void);
 
   AST_Interface *full_definition (void);
+
+  /**
+   * Sets the full definition. If there is an existing dummy definition, it
+   * deletes that first. Ownership of the new definition is NOT assumed.
+   */
   void set_full_definition (AST_Interface *nfd);
 
   virtual bool is_defined (void);
-  void set_as_defined (void);
+
+  /**
+   * Marks this as "defined".
+   *
+   * By default it assumes that ownership of the full definition remains with
+   * this object. If this is not the case, such as if the dummy definition was
+   * redefined and added to a scope, pass true to disown the definition.
+   */
+  void set_as_defined (bool disown = false);
```

Comment:
I'm not too concerned about this since it appears to be an internal API, but it's odd to have a `set_foo` function that takes a single bool but doesn't set the value of `foo` to that bool expression.

Reply:
I also considered a separate `disown()` method, would that be better?